compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -162,12 +162,17 @@ pub(crate) type CompileTimeEvalContext<'mir, 'tcx> =
 #[derive(Debug, PartialEq, Eq, Copy, Clone)]
 pub enum MemoryKind {
     Heap,
+    /// The base allocation of a static.
+    /// When this is read from, we need to error, because otherwise we could
+    /// invent values from nothing (e.g. `static FOO: Thing = FOO;`)
```

Comment:
Is this only for the *current* static or for all of them? The first line indicates the latter, the rest of the comment indicates the former.